Honor store-specific enableOTP setting

shinenelson opened the pull request:
The [Options] allow for the `enableOTP` feature to be specified within
the `.browserpass.json`, but the feature is not enabled when the
option is specified within `.browserpass.json` and not globally.

This was because the check for the setting was only looking for the
global setting - `settings.enableOTP` and did not check for the
`.browserpass.json` setting ( `settings.stores[*].settings.enableOTP` ).

This changeset fixes that issue and properly honors the `enableOTP`
setting specified within `.browserpass.json` with or without the
global `enableOTP` setting.

[Options]: https://github.com/browserpass/browserpass-extension#options

shinenelson committed:
Honor store-specific enableOTP setting

max-baz reviewed (approved):
Thanks, looks good to me, @erayd what do you think?

erayd commented on src/background.js:
```diff
@@ -743,7 +743,7 @@ async function handleMessage(settings, message, sendResponse) {
             break;
         case "copyOTP":
-            if (settings.enableOTP) {
+            if (settings.enableOTP || settings.stores[message.login.store.id].settings.enableOTP) {
```
Please use `getSetting()`.

erayd commented on src/background.js:
```diff
@@ -982,7 +982,10 @@ async function parseFields(settings, login) {
     // preprocess otp
-    if (settings.enableOTP && login.fields.hasOwnProperty("otp")) {
+    if (
+        (settings.enableOTP || settings.stores[login.store.id].settings.enableOTP) &&
```
Please use `getSetting()`.

erayd commented on src/popup/detailsInterface.js:
```diff
@@ -103,7 +103,8 @@ function view(ctl, params) {
                 if (
-                    this.settings.enableOTP &&
+                    (this.settings.enableOTP ||
+                        this.settings.stores[login.store.id].settings.enableOTP) &&
```
Please use the login-scoped settings; it doesn't need to be dereferenced from the higher scope again here.

erayd reviewed (changes requested):
Nice catch, thanks 🙂.

shinenelson committed:
use getSetting() to fetch enableOTP in background.js

also, catch `enableOTP` in `fill` action to copy OTP token after filling login
information

reference : https://github.com/browserpass/browserpass-extension/pull/261#pullrequestreview-632789845

shinenelson committed:
use login-scoped settings in popup/detailsInterface.js

to fetch the `enableOTP` setting

reference : https://github.com/browserpass/browserpass-extension/pull/261#pullrequestreview-632789845

shinenelson commented:
Ping @erayd if the review request was missed

erayd commented:
> Ping @erayd if the review request was missed

Not missed, just short on time. I'm expecting to have some time to spend working on Browserpass during the coming week, which will include this PR.
